src/test/java/org/opensearch/neuralsearch/utils/TestHelper.java

```diff
@@ -0,0 +1,293 @@
+import org.opensearch.common.xcontent.*;
+import org.opensearch.ml.common.FunctionName;
+import org.opensearch.ml.common.dataset.MLInputDataType;
+import org.opensearch.ml.common.dataset.SearchQueryInputDataset;
+import org.opensearch.ml.common.input.MLInput;
+import org.opensearch.ml.common.input.execute.samplecalculator.LocalSampleCalculatorInput;
+import org.opensearch.ml.common.input.parameter.clustering.KMeansParams;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.search.SearchModule;
+import org.opensearch.test.rest.FakeRestRequest;
+
+import com.google.common.collect.ImmutableMap;
+
+public class TestHelper {
+
+    public static final Setting<Boolean> IS_ML_NODE_SETTING = Setting.boolSetting("node.ml", false, Setting.Property.NodeScope);
+
+    public static final DiscoveryNodeRole ML_ROLE = new DiscoveryNodeRole("ml", "ml") {
+        @Override
+        public Setting<Boolean> legacySetting() {
+            return IS_ML_NODE_SETTING;
+        }
+    };
+
+    public static XContentParser parser(String xc) throws IOException {
+        return parser(xc, true);
+    }
+
+    public static XContentParser parser(String xc, boolean skipFirstToken) throws IOException {
+        XContentParser parser = XContentType.JSON.xContent().createParser(xContentRegistry(), LoggingDeprecationHandler.INSTANCE, xc);
+        if (skipFirstToken) {
+            parser.nextToken();
+        }
+        return parser;
+    }
+
+    public static NamedXContentRegistry xContentRegistry() {
+        SearchModule searchModule = new SearchModule(Settings.EMPTY, Collections.emptyList());
+        return new NamedXContentRegistry(searchModule.getNamedXContents());
+    }
+
+    public static String toJsonString(ToXContentObject object) throws IOException {
+        XContentBuilder builder = XContentFactory.jsonBuilder();
+        return xContentBuilderToString(object.toXContent(builder, ToXContent.EMPTY_PARAMS));
+    }
+
+    public static String xContentBuilderToString(XContentBuilder builder) {
+        return BytesReference.bytes(builder).utf8ToString();
+    }
+
+    public static Response makeRequest(
+        RestClient client,
+        String method,
+        String endpoint,
+        Map<String, String> params,
+        String jsonEntity,
+        List<Header> headers
+    ) throws IOException {
+        HttpEntity httpEntity = Strings.isBlank(jsonEntity) ? null : new NStringEntity(jsonEntity, ContentType.APPLICATION_JSON);
+        return makeRequest(client, method, endpoint, params, httpEntity, headers);
+    }
+
+    public static Response makeRequest(
+        RestClient client,
+        String method,
+        String endpoint,
+        Map<String, String> params,
+        HttpEntity entity,
+        List<Header> headers
+    ) throws IOException {
+        return makeRequest(client, method, endpoint, params, entity, headers, false);
+    }
+
+    public static Response makeRequest(
```

Comment:
there are already created functions in OpenSearchRestTestCase class, try to look into them. If you are still not able to find it, move these functions to base class as suggested for upload and loadModel functions, so that all IT can take benefit of.

Reply:
Didn't found the function similar to this makeRequest, moved upload and loadModel and this function to BaseNeuralSearchIT.class